DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -613,8 +613,18 @@ ulong guildId
         };
 
         RestResponse res = await this._rest.ExecuteRequestAsync(request);
+        
+        if (res.ResponseCode == HttpStatusCode.NoContent)
+        {
+            // User was already in the guild, Discord doesn't return the member object in this case
+            return null;
+        }
 
         TransportMember transport = JsonConvert.DeserializeObject<TransportMember>(res.Response!)!;
+        
+        DiscordUser usr = new(transport.User) { Discord = this._discord! };
+
+        this._discord!.UpdateUserCache(usr);
```

Comment:
are we sure we always have a discord client through oauth? worth testing

Reply:
UserCache is on `BaseDiscordClient`